src/FileFormat/PdfStatistics.php

```diff
@@ -0,0 +1,217 @@
+     */
+    protected $imagickDocument;
+
+    /**
+     * Constructor checks, if Imagick extension is available. If yes, instance is created.
+     * @throws \RuntimeException if Imagick extension is not available
+     */
+    public function __construct()
+    {
+        if (!extension_loaded('imagick')) {
+            new \RuntimeException('Imagick php extension is needed to perform PDF statistics.');
+        }
+
+        $this->imagickDocument = new \Imagick();
+    }
+
+    /**
+     * Set filename.
+     * @param string $filename path of pdf file to be analysed
+     */
+    public function setFilename($filename)
+    {
+        $this->imagickDocument->clear();
+        $this->imagickDocument->setResolution(72, 72);
+        $this->imagickDocument->readImage($filename);
+        $this->imagickDocument->setImageUnits(\Imagick::RESOLUTION_PIXELSPERINCH);
+    }
+
+    /**
+     * Method checks, if pdf document pages are in color.
+     * @return array each element is index of page, which is in color (starts from 0)
+     */
+    public function getPagesInColor()
+    {
+        $pagesInColor = array();
+        $pageIterator = 0;
+
+        foreach ($this->imagickDocument as $singlePage) {
+            $histogram = $singlePage->getImageHistogram();
+
+            foreach ($histogram as $pixel) {
+                if (!(
+                    ($pixel->getColorValue(\Imagick::COLOR_RED)==$pixel->getColorValue(\Imagick::COLOR_GREEN)) && 
+                    ($pixel->getColorValue(\Imagick::COLOR_GREEN)==$pixel->getColorValue(\Imagick::COLOR_BLUE)) 
+                )) {    // grayscale is only, when rgb values are the same
+                    $pagesInColor[] = $pageIterator;
+                    break;
+                }
+            }
+
+            $pageIterator++;
+        }
+
+        return $pagesInColor;
+    }
+
+    /**
+     * Function return number of pages in pdf document.
+     * @return integer number of pages
+     */
+    public function getNumberOfPages()
+    {
+        return $this->imagickDocument->getNumberImages();
+    }
+
+    /**
+     * Method identifying page format.
+     * @param string $pageSize page size in 1x1 format
+     * @param boolean $returnPageSizeInMmIfNotMatched if set to false and page size does not fit then function return false (default true)
+     * @return string|boolean name of format; if it does not match: page size or false depending on $returnPageSizeInMmIfNotMatched flag
+     */
+    public function identifyPageSize($pageSize, $returnPageSizeInMmIfNotMatched = true)
+    {
+        if (!isset($this->pageFormats[$pageSize])) {
+            /* format not found, try to rotate it */
+            $pageSizeArray = explode('x', $pageSize);
+            $pageSizeRotated = $pageSizeArray[1] . 'x' . $pageSizeArray[0];
+
+            if (isset($this->pageFormats[$pageSizeRotated])) {
+                $returnPageSize = $this->pageFormats[$pageSizeRotated] . self::PAGE_ROTATED_SUFFIX;
```

Comment:
since `self::PAGE_ROTATED_SUFFIX` is used here, overwriting the constant in a subclass will have no effect.
In case this was not intended, use `static::PAGE_ROTATED_SUFFIX` instead.

Reply:
Is this intended?